### DISCUSSION TITLE
Welcome to email-decomposer Discussions!

### DISCUSSION THREAD

**Discussion by elhanan-mishraky** [intuit/email-decomposer#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sajidhahamed** [comment]
Thanks for the useful module.
Could you please check below,

'ben@bdyatesconstruction.co.uk'

Why this email does not decompose the first name?

**Reply by elhanan-mishraky** [reply]
@sajidhahamed 'ben' is also a dictionary word, this is why it is not extracted as a first name. I'll have to think of a better solution for cases where it is less likely to be a word in the context of an email address. If you have an idea of how to implement it please let me know.

**Reply by elhanan-mishraky** [reply]
@sajidhahamed I fixed this issue, could you please try the latest release (0.0.2)?

**Reply by sajidhahamed** [reply]
@elhanan-mishraky  Thanks for the quick solution. It worked.

**Comment by sajidhahamed** [comment]
Hi Elhanan,

'simon.cawte@cr.co.uk' in this mail id last name is not extracted.
Could you please check why?

**Reply by sajidhahamed** [reply]
Also ['sales@affinity-homes.co.uk']
This email got the last name as 'sales'

'team@baileyconstructioncivils.co.uk'
here 'team' came as a last name

**Reply by elhanan-mishraky** [reply]
Hi @sajidhahamed,
Cawte does not appears on census last name file which I use, if you have another source of last names please let me know.
As for sales/team I will try to fix it soon.

**Comment by elhanan-mishraky** [comment]
@sajidhahamed, can you try the latest release? It should solve the sales/team issue.

**Reply by sajidhahamed** [reply]
It worked. But if we have trailing or leading space this fails. So better strip the string and pass into your algorithm.
Thanks again for the fix
![Screenshot 2023-02-17 161612](https://user-images.githubusercontent.com/48512874/219622882-1f7e96e7-fb27-4d27-9bc2-2697efdcd8a5.png)
